src/shared/ui/InputIcon.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+
+import { IconProps } from 'react-native-vector-icons/Icon';
+
+export type InputIconProps<T> = Partial<T> & {
+  iconProps?: Partial<IconProps>;
+  input: React.ReactNode;
+  icon: React.ReactNode;
+};
+
+const styles = StyleSheet.create({
+  inputContainer: {
+    justifyContent: 'center',
+  },
+  input: {
+    height: 'auto',
+  },
+  icon: {
+    position: 'absolute',
+    right: 10,
+  },
+});
+
+export function InputIcon<T = any>({ input, icon }: InputIconProps<T>) {
+  return (
+    <View style={styles.inputContainer}>
+      <View style={styles.input}>{input}</View>
+      <View style={styles.icon}>{icon}</View>
+    </View>
```

Comment:
Just a few little things to mention:

- I think we should take advantage of Tamagui components whenever possible, as it will reduce the amount of styles we have to write as well as put us in a better position for eventually unifying the UI component library across FE apps. I haven't seen this made into a best practice document yet, but I'd like to push for it.
- Generally, I'd avoid absolutely positioning an icon contained inside an inputs, and use natural flexbox layout instead, else we tend to run into situations like this… 😏
  <img width="414" alt="Screenshot 2024-02-01 at 9 02 17 PM" src="https://github.com/ChildMindInstitute/mindlogger-app-refactor/assets/1282772/94806333-7ae6-4655-83a6-2f04e2b31a62">
- However, the solution I had in mind:
  ```suggestion
    <XStack alignItems="center">
      <Box flex={1}>{input}</Box>
      <Box px={10}>{icon}</Box>
    </XStack>
  ```
  doesn't _quite_ solve it because the bottom border focus effect is being applied only inside the `input` and not _this_ higher level component. 😔 So a perfect solution would probably require integrating the right icon as an optional prop on the `InputField` component, and placing the above `XStack` in there.